frontend/packages/kubevirt-plugin/integration-tests/tests/vm.resources.scenario.ts

```diff
@@ -82,45 +90,40 @@ describe('Add/remove disks and NICs on respective VM pages', () => {
     VM_ACTIONS_TIMEOUT_SECS,
   );
 
-  it('ID(CNV-1722) NIC cannot be added twice using one net-attach-def', async () => {
-    await vm.navigateToTab(TAB.NetworkInterfaces);
-    if (
-      (await vm.getAttachedNICs()).filter((nic) => nic.name === multusNetworkInterface.name)
-        .length === 0
-    ) {
+  it(
+    'ID(CNV-1722) NIC can be added twice using one net-attach-def',
+    async () => {
+      const secondMultusInterface = {
+        name: `nic2-${testName.slice(-5)}`,
+        model: NIC_MODEL.VirtIO,
+        mac: getRandomMacAddress(),
+        type: NIC_TYPE.bridge,
+        network: multusNAD.metadata.name,
+      };
       await vm.addNIC(multusNetworkInterface);
-    }
-
-    // Verify the NIC is added in VM Manifest
-    expect(
-      _.find(getInterfaces(vm.getResource()), (o) => o.name === multusNetworkInterface.name),
-    ).toBeDefined();
-
-    // Try to add the NIC again
-    await click(createNICButton, 1000);
-    await browser.sleep(1000).then(() => browser.wait(until.presenceOf(nicNetwork)));
-
-    // The network dropdown should be either empty (disabled) or not containing the already used net-attach-def
-    await browser.wait(
-      until.or(
-        async () => {
-          return !(await nicNetwork.isEnabled());
-        },
-        async () => {
-          return !(await getSelectOptions(nicNetwork)).includes(multusNetworkInterface.network);
-        },
-      ),
-    );
-  });
+      await vm.addNIC(secondMultusInterface);
+      expect(await vm.getAttachedNICs()).toContain(multusNetworkInterface);
+      expect(await vm.getAttachedNICs()).toContain(secondMultusInterface);
+      await vm.action(VM_ACTION.Start);
+      const vmInterfaces = getInterfaces(vm.getResource());
+      expect(
+        _.filter(
+          vmInterfaces,
+          (o) => o.name === multusNetworkInterface.name && o.name === secondMultusInterface.name,
+        ),
+      ).toBeDefined();
```

Comment:
Careful when using `toBeDefined`. Method `find` returns either found element or `undefined`. However, `filter` returns an array, which mean the `toBeDefined` with `filter` will always return true. Even when the array is empty, which it is in this case because of `&&`.

```
const vmInterfaces = _.filter(
        getInterfaces(vm.getResource()),
        (o) => o.name === multusNetworkInterface.name || o.name === secondMultusInterface.name,
      );
expect(vmInterfaces.length).toEqual(2);
```

Reply:
updated now.